File: py/adapters/lora_manager.py
from .base import BaseAdapter


class LoraManagerAdapter(BaseAdapter):
    adapter_name = "lora_manager"

    def can_read(self, node_info: dict) -> bool:
        return node_info.get("comfy_class") in {
            "Lora Loader (LoraManager)",
            "Lora Stacker (LoraManager)",
            "Prompt (LoraManager)",
        }

File: web/js/adapters/lora_manager.js
import { TextSyntaxAdapter } from "./text_syntax.js";

const LM_CLASSES = new Set([
  "Lora Loader (LoraManager)",
  "Lora Stacker (LoraManager)",
  "WanVideo Lora Select (LoraManager)",
]);

export class LoraManagerAdapter extends TextSyntaxAdapter {
  name = "lora_manager";

  canRead(node) {
    return LM_CLASSES.has(node?.comfyClass || "") && super.canRead(node);
  }

  canApplyLora(node) {
    return LM_CLASSES.has(node?.comfyClass || "") && super.canApplyLora(node);
  }
}
